ephios/api/views/events.py

```diff
@@ -96,3 +96,11 @@ class EventViewSet(viewsets.ReadOnlyModelViewSet):
         .prefetch_related(Prefetch("shifts__participations"))
         .order_by("start_time")
     )
+
+
+class ParticipationSerializer(serializers.ModelSerializer):
+    state = serializers.CharField(source="get_state_display")
+
+    class Meta:
+        model = LocalParticipation
+        fields = ["id", "shift", "state", "comment"]
```

Comment:
@felixrindt I wanted to include `start_time` and `end_time` as well, but this fails. Do you have an idea what I need to do to make it work? as it should use our custom manager?

Reply:
Did you even look at what ModelSerializer does? :D
It checks for properties on the model class, which we obvsl don't have if the value comes from a custom manager. We can however hook into the field creation process and just tell it to treat it like a property field:



```suggestion
        fields = ["id", "shift", "state", "comment", "start_time", "end_time"]

    def build_unknown_field(self, field_name, model_class):
        if field_name in {"start_time", "end_time"}:
            return self.build_property_field(field_name, model_class)
        return super().build_unknown_field(field_name, model_class)
```